extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java

```diff
@@ -161,38 +154,39 @@ public static class Secret {
 
             public static enum Method {
                 /**
-                 * client_secret_basic (default): client id and secret are submitted with the HTTP Authorization Basic scheme
+                 * `client_secret_basic` (default): The client id and secret are submitted with the HTTP Authorization Basic
+                 * scheme.
                  */
                 BASIC,
 
                 /**
-                 * client_secret_post: client id and secret are submitted as the `client_id` and `client_secret` form
-                 * parameters.
+                 * `client_secret_post`: The client id and secret are submitted as the `client_id` and `client_secret`
+                 * form parameters.
                  */
                 POST,
 
                 /**
-                 * client_secret_jwt: client id and generated JWT secret are submitted as the `client_id` and `client_secret`
-                 * form
-                 * parameters.
+                 * `client_secret_jwt`: The client id and generated JWT secret are submitted as the `client_id` and
+                 * `client_secret`
+                 * form parameters.
                  */
                 POST_JWT
             }
 
             /**
-             * The client secret value - it will be ignored if 'credentials.secret' is set
+             * Optional. The client secret value. This value is ignored if `credentials.secret` is set.
              */
             @ConfigItem
             public Optional<String> value = Optional.empty();
 
             /**
-             * The Secret CredentialsProvider
+             * The Secret CredentialsProvider.
              */
             @ConfigItem
             public Provider provider = new Provider();
 
             /**
-             * Authentication method.
+             * Optional. The authentication method.
```

Comment:
I think Optional does not work here, do not recall why. But if the `clientSecret.value` secret is set then this method will be basic by default - may be this is what should be said. There are different variations which is why it is optional.

Reply:
Updated. Thanks. Is this what you're suggesting?
```suggestion
             * The authentication method.
             * If the `clientSecret.value` secret is set, this method is `basic` by default.
```

Reply:
I believe this conversation is resolved. Please reopen it if needed.